model/DataObject.php

```diff
@@ -3421,7 +3421,12 @@ public function databaseIndexes() {
 						continue;
 					}
 
-					if ($this->hasOwnTableDatabaseField($column) && !array_key_exists($column, $indexes)) {
+					list($fieldType) = Object::parse_class_spec($this->db($column));
+					if (
+						$this->hasOwnTableDatabaseField($column)
+						&& !array_key_exists($column, $indexes)
+						&& $fieldType !== 'Text' && !is_subclass_of($fieldType, 'Text')
```

Comment:
I'd prefer a less hard-coded approach here, where we can set a config list of blacklisted index types?

Reply:
Yeah sure. Do you think `DataObject` makes sense, or should it be a setting like `DBField.auto_indexable` and then set `private static $auto_indexable = false;` on `Text`?

Reply:
oooh, I like the latter idea 👍